docs/plugins.md

```diff
@@ -7,190 +7,177 @@ layout: layouts/extend.njk
+    // return {
+    //   '.js': jsFileContents,
+    //   '.js.map': jsMapContents,
+    //   '.css': cssFileContents'
+    //   …
+    // };
+  }
+})
 ```
 
-Bundle the web application for production.
+### Build your own
 
+#### Local plugin
 
-👉 **[Back to the main docs.](/)**
+If you need a local plugin just for your own needs, add it in your `snowpack.config.json`:
 
-## TypeScript types
+```json
+{
+  "plugins": [
+    ["../path-to-my-plugin", { "optionA": "foo", "optionB": "bar" }]
+  ]
+}
+```
 
-Building your Snowpack plugin with TypeScript? You can typecheck your plugin by importing the following type:
+#### npm plugin
 
-```ts
-import { SnowpackPlugin } from 'snowpack/dist-types/config';
+To release a plugin to npm, take a look at [snowpack-plugin-starter-template](https://github.com/pikapkg/snowpack-plugin-starter-template) which can get you up-and-running quickly. You can either copy this outright, or simply take what you need.
 
-export default function mySnowpackPlugin() {
-  const plugin: SnowpackPlugin = { /* … */ };
-  return plugin;
-}
-```
+In general, make sure to mind the following checklist:
 
-## JavaScript API
+- ✔️ Your `package.json` file has a `main` entry pointing to the final build
+- ✔️ Your code is compiled to run on Node >= 10
+- ✔️ Your package README contains a list of custom options, if your plugin is configurable
 
-### Overview
+### Tips / Gotchas
 
-```js
-// TODO: More documentation coming soon.
-```
+- Snowpack’s plugin system is **single-pass** to keep things speedy and avoid infinite loops. That means if you generate a new `.css` file, that file won’t be run back through another CSS plugin.
+- Extensions in Snowpack always have a leading `.` character (e.g. `.js`, `.ts`). This is to match Node’s `path.extname()` behavior, as well as make sure we’re not matching extension substrings (e.g. if we matched `js` at the end of a file, we also don’t want to match `.mjs` files by accident; we want to be explicit there).
+- You may have seen a `transform()` function in previous versions of the plugin API. Though that’ll be supported throughout Snowpack `2.x`, it’s now deprecated and will be removed in the next major release. Instead, please use `build()`.
+- The `input` and `output` file extension arrays are the keys to their simplicity. In other plugin systems, you have to parse out the filename to determine whether you want to transform it or not. But with Snowpack, you only have to declare file extensions you’d like to listen for, and Snowpack handles the rest.
+- If `build()` doesn’t return anything, the file isn’t transformed. In this way, you can build pure side-effect plugins if needed.
+- If you want to build a plugin that only runs some code on initialization (such as `@snowpack/plugin-dotenv`), put your side-effect code inside your main plugin function to execute when initialized. But be sure to return a `{ name }` so that in case of error, users know what went wrong.
```

Comment:
super nit:

```suggestion
- If you want to build a plugin that only runs some code on initialization (such as `@snowpack/plugin-dotenv`), put your side-effect code inside your main plugin function to execute when initialized. But be sure to return a `{ name }` so that users know what went wrong in the case of an error.
```

Reply:
+1. I’m a fan of active voice as well for clarity.